### PR TITLE
Fix startup on IE10

### PR DESCRIPTION
**Pull request by Rosch**
Current Cappuccino apps do not startup on IE10.
(They startup on IE11 in IE10 quirks mode but not in native IE10).

As I *NEED* to run there I digged into this issue.
Please be graceful to me: I am not a javascript and IE Guru - but I found the problem.

IE10 does not allow to set properties on an XMLHttpRequest instance if it is not open.
I altered CFHTTPRequest.js to postpone the particular withCredentials setting after the request has been opened.

And voila - my capp apps start on IE10 just fine. Regressen tests on other browser did not reveal a problem.

Here is my contribution back to the project to fix this issue.

PS: This is also my very first pull-request ever so I hope I did not make anything foolish...

### CODE REVIEW

**Comment by ahankinson** [comment]
Did you test this with other browsers? There is a CORS test in Tests/Manual/CrossOrigin with a readme on how to run the little python server to test that `withCredentials` still works.

You might also have a look at this note:

https://github.com/cappuccino/cappuccino/blob/master/Tests/Manual/CrossOriginTest/README.md#a-note-about-ie

**Comment by Rosch** [comment]
FIRST: Thank you for the fast reaction on my change!
I am quite new to all these github things. But I am learning... Sorry for inconvenience.

Back to my patch/pull request: 
I reformatted it to follow the 4 spaces - no tab rules. Thanks for the hint.

After looking at the issues of Cappuccino: I see this in relation to issues #2157 / #2158 / #2272. Here the changes to CFHTTPRequest where made for better IE compatability but the throwed up the current issue with native IE10. So I see this as an addendum to the changes there.

About your concern regarding the setting of `withCredentials`. Yes it is a little delayed. I grepped thru the source. The only real places where this concerns is CPURLConnection. There are 2 places where the `setWithCredentials:` is called. In one case (`+sendSynchronousRequest:`) the connection is opened right after the `setWithCredentials:` call. So it happens IMHO also immediately.

The second case is `-initWithRequest:` Here it can be immediately, too. But not necessarily. But even it is not performed immediately it happens when the connection is going to work. In my opinion this does not harm. An error would arise by then and is not eaten up.

Hope this helps to clear this up...

**Comment by primalmotion** [comment]
Hey, can you merge the master back in this one? Thanks

**Comment by Rosch** [comment]
Hi Antoine...

You are meaning me I assume... Sorry for stupid questions. I am still a "github baby".

I shall merge the current branch of cappuccino into my copy to update the pull request? Right? 
The conflict must have come from the recent commits in this area...

**Comment by primalmotion** [comment]
@Rosch hey,

You need to do on your local fork

```
git checkout master
git pull upstream master #if the official git repo is named `upstream`
.. fix the conflicts ...
git push origin master # to update the PR
```

As a side note, you should not work on the master branch of your fork of Cappuccino. You should have created a new branch named `fix-ie10-loading` from the latest master, and open a PR from this branch. It's no big deal, but it's a good practice to have. Imagine you work on 10 differents new things, you can't fit them all in your maste branch :)

**Comment by primalmotion** [comment]
there's a a pretty good doc about the github workflow here https://guides.github.com/introduction/flow/index.html

**Comment by Rosch** [comment]
Hi Antoine...

Sorry for the delay. I had a terrible weekend. I merged now the current master to my master.
Everything appears to be fime. I did no additional changes to `Objective-J/CFHTTPRequest.js`.

**Comment by Rosch** [comment]
Hi....

A made an addendum to my pull request. There was a warning in chrome that synchronous XHR requests using Credentials are deprecated. Also FF 40 seems to be more picky about this now, too.

My last change on this fixes this also.

File: Objective-J/CFHTTPRequest.js [hunk 1]
```diff
@@ -237,13 +240,23 @@ CFHTTPRequest.prototype.overrideMimeType = function(/*String*/ aMimeType)
...
+
     this._isOpen = true;
     this._URL = aURL;
     this._async = isAsynchronous;
     this._method = aMethod;
     this._user = aUser;
     this._password = aPassword;
-    return this._nativeRequest.open(aMethod, aURL, isAsynchronous, aUser, aPassword);
+    
+    retval = this._nativeRequest.open(aMethod, aURL, isAsynchronous, aUser, aPassword);
```
Comment (Dogild) [inline Objective-J/CFHTTPRequest.js]: Add a line break here

File: Objective-J/CFHTTPRequest.js [hunk 2]
```diff
@@ -217,12 +218,14 @@ CFHTTPRequest.prototype.getResponseHeader = function(/*String*/ aHeader)
 
 CFHTTPRequest.prototype.setTimeout = function(/*int*/ aTimeout)
 {
-    this._nativeRequest.timeout = aTimeout;
+    this._timeout = aTimeout;
+    if (this._isOpen)
```
Comment (Dogild) [inline Objective-J/CFHTTPRequest.js]: You should check if it's a async request or not here

**Comment by Dogild** [comment]
@primalmotion looks like the problem we had with phanotmjs yesterday will be fixed with this pr

**Comment by Rosch** [comment]
So maybe its time to merge?

We are running here cappuccino with this patch in place since months without any problem on many different browser versions

**Comment by Dogild** [comment]
Yep, could you only fix the small remarks I commented ? Then it will be ready to merge

**Comment by Dogild** [comment]
Thanks !

**Pull request merged**